src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -134,13 +136,52 @@ public static WwwAuthenticateParameters CreateFromWwwAuthenticateHeaderValue(str
         /// Create the authenticate parameters by attempting to call the resource unauthenticated, and analyzing the response.
         /// </summary>
         /// <param name="resourceUri">URI of the resource.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
         /// <returns>WWW-Authenticate Parameters extracted from response to the un-authenticated call.</returns>
-        public static async Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(string resourceUri)
+        public static Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(string resourceUri, CancellationToken cancellationToken = default)
         {
+            var httpClientFactory = PlatformProxyFactory.CreatePlatformProxy(null).CreateDefaultHttpClientFactory();
+            return CreateFromResourceResponseAsync(httpClientFactory, resourceUri, cancellationToken);
+        }
+
+        /// <summary>
+        /// Create the authenticate parameters by attempting to call the resource unauthenticated, and analyzing the response.
+        /// </summary>
+        /// <param name="httpClientFactory">Factory to produce an instance of <see cref="HttpClient"/> to make the request with.</param>
+        /// <param name="resourceUri">URI of the resource.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+        /// <returns>WWW-Authenticate Parameters extracted from response to the un-authenticated call.</returns>
+        public static Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(IMsalHttpClientFactory httpClientFactory, string resourceUri, CancellationToken cancellationToken = default)
+        {
+            if (httpClientFactory is null)
+            {
+                throw new ArgumentException($"'{nameof(httpClientFactory)}' cannot be null.", nameof(httpClientFactory));
```

Comment:
nit: We typically use ArgumentNullException for these types of checks on the api surface.

Reply:
Yeah, I noticed that too. Let me explain.

The only already existing check is for resourceId for not null or empty or white space. And it throws just `ArgumentException`. Some people may argue that if the input is an empty string the code can't throw `ArgumentNullException` because it's not null. But I personally think that's nitpicking and matters less than readable and consistent code.

In this PR I'm adding two more checks, both are for reference types so I'm just checking for not null.
And the caveat is in the order of parameters: `ArgumentException(message, paramName)` vs `ArgumentNullException(paramName, message)`. Why the inconsistency is a question to BCL. So in our case the code looks little bit more messy and less readable.

So what I would do is to throw `ArgumentNullException` in all checks. What you think?

Reply:
Yes, we do that as well in most places, i.e. throw `ArgumentNullException` for empty string. 

Strings should really not be nullable :), but unfortunately we are not a library that uses the nullable references feature, so we have to live with these small inconsistencies.